mobile/mobile.go

```diff
@@ -0,0 +1,505 @@
+	lndConn *grpc.ClientConn
+
+	statusChecker func() mailbox.ClientStatus
+
+	mac *macaroon.Macaroon
+
+	registry map[string]func(context.Context, *grpc.ClientConn,
+		string, func(string, error))
+}
+
+type stubPackageRegistration func(map[string]func(context.Context,
+	*grpc.ClientConn, string, func(string, error)))
+
+type NativeCallback interface {
+	SendResult(json string)
+}
+
+var (
+	registrations = []stubPackageRegistration{
+		lnrpc.RegisterLightningJSONCallbacks,
+		lnrpc.RegisterStateJSONCallbacks,
+		autopilotrpc.RegisterAutopilotJSONCallbacks,
+		chainrpc.RegisterChainNotifierJSONCallbacks,
+		invoicesrpc.RegisterInvoicesJSONCallbacks,
+		routerrpc.RegisterRouterJSONCallbacks,
+		signrpc.RegisterSignerJSONCallbacks,
+		verrpc.RegisterVersionerJSONCallbacks,
+		walletrpc.RegisterWalletKitJSONCallbacks,
+		watchtowerrpc.RegisterWatchtowerJSONCallbacks,
+		wtclientrpc.RegisterWatchtowerClientJSONCallbacks,
+		looprpc.RegisterSwapClientJSONCallbacks,
+		poolrpc.RegisterTraderJSONCallbacks,
+		frdrpc.RegisterFaradayServerJSONCallbacks,
+	}
+
+	perms = core.GetAllMethodPermissions()
+
+	jsonCBRegex = regexp.MustCompile(`(\w+)\.(\w+)\.(\w+)`)
+
+	m *mobileClient
+
+	registry = make(map[string]func(context.Context,
+		*grpc.ClientConn, string, func(string, error)))
+
+	localPrivCreateCallback  NativeCallback
+	remoteKeyReceiveCallback NativeCallback
+	authDataCallback         NativeCallback
+)
+
+// InitLNC sets up everything required for LNC to run including
+// signal interceptor, logs, and an instance of the mobile client.
+func InitLNC(debugLevel string) {
+	defer func() {
+		if r := recover(); r != nil {
+			fmt.Printf("Recovered in f: %v", r)
+			debug.PrintStack()
+		}
+	}()
+
+	// set debug level to 'info' if not specified
+	if debugLevel == "" {
+		debugLevel = "info"
+	}
+
+	// Hook interceptor for os signals.
+	shutdownInterceptor, err := signal.Intercept()
+	if err != nil {
+		exit(err)
+	}
+
+	logWriter := build.NewRotatingLogWriter()
+	SetupLoggers(logWriter, shutdownInterceptor)
+
+	err = build.ParseAndSetDebugLevels(debugLevel, logWriter)
+	if err != nil {
+		exit(err)
+	}
+
+	m = newMobileClient()
+
+	for _, registration := range registrations {
+		registration(registry)
+	}
+
+	log.Debugf("Mobile client ready for connecting")
+}
+
+func newMobileClient() *mobileClient {
+	return &mobileClient{
+		lndConn: nil,
+		registry: make(map[string]func(context.Context,
+			*grpc.ClientConn, string, func(string, error))),
+	}
+}
+
+// ConnectServer creates a connection from the client to the
+// mailbox server.
+func ConnectServer(mailboxServer string, isDevServer bool,
+	pairingPhrase string, localStatic string, remoteStatic string) {
+	if mailboxServer == "" {
+		exit(fmt.Errorf("invalid use of ConnectServer, " +
+			"need parameter mailboxServer"))
+	}
+
+	// Check that the correct arguments and config combinations have been
+	// provided.
+
+	err := validateArgs(localStatic, remoteStatic)
+	if err != nil {
+		exit(err)
```

Comment:
im assuming it is for mobile-specific reasons that we dont return the errors in this function?

Reply:
This is actually something carried over from the wasm code. I'll refactor this.